library/network/src/NetworkInterfaces.ycp

```diff
@@ -226,25 +349,64 @@ global string GetTypeFromIfcfg(ifcfg_t ifcfg) {
 	  type = rule_type;
 	}
       });
+
     foreach(string rule_key, TypeByKeyValue, {
 	string rule_type = ifcfg[rule_key]:"";
 	if (rule_type != "") {
 	  type = rule_type;
 	}
       });
-//y2internal("from ifcfg: %1", type);
+
     return type;
 }
 
-global string GetType(string dev) {
-    string type = device_type (dev);
-    foreach(string dev_type,map<string, ifcfg_t> confs, Devices, {
+/*
+ * Detects device type according its name and ifcfg configuration.
+ *
+ * @param dev   device name
+ * @param ifcfg device's ifcfg configuration
+ * @return      device type
+ */
+string GetTypeFromIfcfgOrName( string dev, ifcfg_t ifcfg)
+{
+    if( IsEmpty( dev) )
+        return nil;
+
+    string type = GetTypeFromSysfs( dev);
+
+    if( IsEmpty( type) )
+        type = GetTypeFromIfcfg( ifcfg);
+
+    if( type == nil)
+        type = device_type( dev);
+
+    y2milestone( "GetTypeFromIfcfgOrName: device='%1', type='%2'", dev, type);
+
+    return type;
+}
+
+/*
+ * Detects device type according cached data
+ *
+ * If cached ifcfg for given device is found it is used as parameter for
+ * GetTypeFromIfcfgOrName( dev, ifcfg). Otherwise is device handled as unconfigured
+ * and result is equal to GetTypeFromIfcfgOrName( dev, nil)
+ *
+ * @param dev   device name
+ * @return      detected device type
+ */
+global string GetType(string dev) 
+{
+    string type = GetTypeFromIfcfgOrName( dev, nil);
+
+    foreach(string dev_type,map<string, ifcfg_t> confs, Devices, 
+    {
 	ifcfg_t ifcfg = confs[dev]:$[];
-	string ifcfg_type = GetTypeFromIfcfg(ifcfg);
-	if (ifcfg_type != nil)
-	  type = ifcfg_type;
-      });
-//y2internal("original %1: %2", dev, type);
+        
+        if( !IsEmpty( ifcfg) )
+            type = GetTypeFromIfcfgOrName( dev, ifcfg);
```

Comment:
Seems so ... but no. This code is consequence of complexity in structure where cached configuration is stored. It in fact only tries to find cached sysconfig configuration for particular device name.